enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/pipes/UnionSlottedPipe.scala

```diff
@@ -45,12 +36,8 @@ case class UnionSlottedPipe(lhs: Pipe, rhs: Pipe, lhsInfo: PipelineInformation,
     new Iterator[ExecutionContext] {
       override def hasNext: Boolean = left.hasNext || right.hasNext
 
-      override def next(): ExecutionContext = if (left.hasNext) {
-        val in = left.next()
-        val out = PrimitiveExecutionContext(unionInfo)
-        mapSlots.foreach(f => f(in, out, lhsInfo, state))
-        out
-      } else {
+      override def next(): ExecutionContext = if (left.hasNext) left.next()
```

Comment:
This is wrong in the general case the left-hand side also might need to be transformed